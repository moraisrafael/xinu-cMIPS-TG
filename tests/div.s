	.include "cMIPS.s"
	.text
	.align 2
	.globl _start
	.ent _start
_start: nop
	la   $15,x_IO_BASE_ADDR
	li   $3,3
	li   $4,3
rep:	sw   $4, 0($15)           # IO[0]=3,5,7,9,11,13,15
	div  $0,$4,$3
	nop
	nop
	mflo $5
	sw   $5, 0($15)  # QUOC IO[0]=1,1,2,3,3,4,5
	mfhi $6
	sw   $6, 4($15)  # REM  IO[60]=0,2,1,0,2,1,0
	addi $4,$4,2
	slti $22,$4,16
	bne  $22,$0,rep
	nop
	nop
	nop
	nop
	wait
	.end _start
