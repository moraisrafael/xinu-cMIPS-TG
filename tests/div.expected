00000003
00000001
00000000
00000005
00000001
00000002
00000007
00000002
00000001
00000009
00000003
00000000
0000000b
00000003
00000002
0000000d
00000004
00000001
0000000f
00000005
00000000
